tcpdump.c

```diff
@@ -1162,17 +1162,26 @@ _U_
 		free(host_url);
 	} else
 #endif
-	status = pcap_findalldevs(&devlist, ebuf);
-	if (status < 0)
-		error("%s", ebuf);
+        status = pcap_findalldevs(&devlist, ebuf);
+        if (status < 0) {
+  		      if (devlist != NULL) {
```

Comment:
Still needs the indentation adjusted.

Reply:
Also, note that if `pcap_findalldevs()` returns an error indication, `devlist` will be null.